pkg/trtl/honu_test.go

```diff
@@ -0,0 +1,330 @@
+		require.Equal(requests[e.Id].Id, e.Id)
+		require.Contains(e.Error, "not implemented")
+	}
+}
+
+func (s *trtlTestSuite) TestIter() {
+	require := s.Require()
+	ctx := context.Background()
+
+	// Start the gRPC client.
+	conn, err := s.connect()
+	require.NoError(err)
+	defer conn.Close()
+	client := pb.NewTrtlClient(conn)
+
+	// Test cannot use reserved namespace
+	_, err = client.Iter(ctx, &pb.IterRequest{Namespace: "index"})
+	s.StatusError(err, codes.PermissionDenied, "cannot used reserved namespace")
+
+	// Test Invalid Options
+	_, err = client.Iter(ctx, &pb.IterRequest{Namespace: "people", Options: &pb.Options{IterNoKeys: true, IterNoValues: true}})
+	s.StatusError(err, codes.InvalidArgument, "cannot specify no keys, values, and no return meta: no data would be returned")
+
+	// Test iter default prefix, default options, expecting 1 response from default namespace
+	rep, err := client.Iter(ctx, &pb.IterRequest{})
+	require.NoError(err, "could not iterate default prefix with default options")
+	require.Len(rep.Values, 1, "too many responses returned, did the fixtures change?")
+	require.Empty(rep.NextPageToken, "a next page token was returned for a one page response")
+	require.NotEmpty(rep.Values[0].Key, "key not supplied in iter by default")
+	require.NotEmpty(rep.Values[0].Value, "value not supplied in iter by default")
+	require.Equal("default", rep.Values[0].Namespace, "non-default namespace fetched")
+	require.Empty(rep.Values[0].Meta, "meta returned by default")
+
+	// Test invalid page token
+	token := "CAISLHBlb3BsZTo6NDZlNzg5MTctOGQyMC00N2MwLWIwZDEtZTUyMDQxNDlhOTM2"
+	_, err = client.Iter(ctx, &pb.IterRequest{Options: &pb.Options{PageToken: "foo"}})
+	s.StatusError(err, codes.InvalidArgument, "invalid page token")
+	_, err = client.Iter(ctx, &pb.IterRequest{Namespace: "people", Options: &pb.Options{PageToken: token, PageSize: 27}})
+	s.StatusError(err, codes.InvalidArgument, "page size cannot change between requests")
+	_, err = client.Iter(ctx, &pb.IterRequest{Namespace: "things", Options: &pb.Options{PageToken: token, PageSize: 2}})
+	s.StatusError(err, codes.InvalidArgument, "prefix and namespace cannot change between requests")
+	_, err = client.Iter(ctx, &pb.IterRequest{Prefix: []byte("zed"), Namespace: "people", Options: &pb.Options{PageToken: token, PageSize: 2}})
+	s.StatusError(err, codes.InvalidArgument, "prefix and namespace cannot change between requests")
+
+	// Test ordered non-paginated request with prefix
+	rep, err = client.Iter(ctx, &pb.IterRequest{Namespace: "people", Prefix: []byte("213")})
+	require.NoError(err, "could not fetch complete iteration")
+	require.Empty(rep.NextPageToken, "a next page token was returned for a one page response")
+	require.Len(rep.Values, 5, "incorrect responses returned did the fixtures change?")
+
+	expectedOrder := []string{
+		"alice", "bob", "charlie", "darelene", "erica",
+		"franklin", "gregor", "helen", "ivan", "juliet",
+	}
+
+	for i := 0; i < 5; i++ {
```

Comment:
Expected order is failing, which is also causing pagination to fail … not sure what's going on; could be something in Honu or the way the fixtures are inserted. More investigation required.